syft/pkg/cataloger/cpp/parse_conanlock.go

```diff
@@ -0,0 +1,62 @@
+package cpp
+
+import (
+	"encoding/json"
+	"io"
+	"strings"
+
+	"github.com/anchore/syft/syft/artifact"
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/pkg/cataloger/common"
+)
+
+// integrity check
+var _ common.ParserFn = parseConanlock
+
+type conanLock struct {
+	GraphLock struct {
+		Nodes map[string]struct {
```

Comment:
nit: this could be `map[string]pkg.ConanMetadata` instead of redefining these. That is, if the metadata is meant to raise up the raw packaging metadata, if we remove name/version then there would be no difference between this struct def and `pkg.ConanMetadata`.

Reply:
I could be wrong about this if `options string` should really be transformed into `options []string` (split by newline) or `options map[string]string` if split by newline and `=`. (same comment for similar fields)

Reply:
For those following along here are all possible fields for the conan.lock structure
https://github.com/conan-io/conan/blob/develop/conans/model/graph_lock.py#L101-L276